cfme/cloud/instance/__init__.py

```diff
@@ -38,33 +35,116 @@
-            return title_match and prov_match
+            return InfoBlock.text('Relationships', 'Cloud Provider') == provider_name
         except BlockTypeUnknown:
             # Default to false since we can't identify which provider the image belongs to
             return False
 
 
-@VM.register_for_provider_type("cloud")
+class CloudInstanceView(BaseLoggedInPage):
+    title = Text('#explorer_title_text')
+
+    configuration = Dropdown('Configuration')
+    lifecycle = Dropdown('Lifecycle')
+    policy = Dropdown('Policy')
+    power = Dropdown('Power')
+
+    @property
+    def in_cloud_instance(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Compute', 'Clouds', 'Instances']
+        )
+
+    @View.nested
+    class instances_by_provider(Accordion): # noqa
+        ACCORDION_NAME = 'Instances by Provider'
+        tree = ManageIQTree()
+
+    @View.nested
+    class images_by_provider(Accordion): # noqa
+        ACCORDION_NAME = 'Images by Provider'
+        tree = ManageIQTree()
+
+    @View.nested
+    class instances(Accordion): # noqa
+        ACCORDION_NAME = 'Instances'
+        tree = ManageIQTree()
+
+    @View.nested
+    class images(Accordion): # noqa
+        ACCORDION_NAME = 'Images'
+        tree = ManageIQTree()
+
+
+class InstanceAllView(CloudInstanceView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_cloud_instance and
+            self.title.text == 'All Instances'
+        )
+
+
+class InstanceByProviderView(CloudInstanceView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_cloud_instance and
+            self.title.text == 'All Instances by Provider'
+        )
+
+
+class InstanceDetailsView(CloudInstanceView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_cloud_instance and
+            (self.title.text == 'Instance "{}"'.format(self.obj.name) and
+                details_page_check(self.obj.provider.name))
+        )
+
+
+class InstanceProvisionForm(ProvisioningForm):
+    image_table = Table('//div[@id="pre_prov_div"]//table')
+    continue_button = Button('Continue')  # Continue button on 1st page, image selection
+    submit_button = Button('Submit')  # Submit for 2nd page, tabular form
+    cancel_button = Button('Cancel')  # Abort!
+
+
+class InstanceProvisionView(InstanceProvisionForm):
+    title = Text('#explorer_title_text')
+
+    @property
+    def is_displayed(self):
+        # TODO: Match against image selected in Catalog tab
+        return False
+
+    def before_fill(self, values):
+        # Fill first part of form 'manually' here
+        # fingers crossed this works like I'm planning
+        try:
+            row = self.image_table.row(Name=self.context['object'].template_name,
+                                       Provider=self.context['object'].provider.name)
```

Comment:
When we're using the keywords, I would stick to the `snake_cased` with all-lowercase to keep in line with python style.